assets/js/googlesitekit/datastore/user/permissions.js

```diff
@@ -315,6 +315,29 @@ const baseSelectors = {
 			);
 		}
 	),
+
+	/**
+	 * Checks if the current user has access to the specified shareable module.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param {Object} state      Data store's state.
+	 * @param {string} moduleSlug Module slug to check.
+	 * @return {boolean} `true` if the user is authenticated or if the module is shareable and viewable by the current user. `false` if the module does not exist, is not shareable or not viewable by the current user.
+	 */
+	hasAccessToShareableModule: createRegistrySelector(
+		( select ) => ( state, moduleSlug ) => {
+			if ( ! select( CORE_MODULES ).isModuleAvailable( moduleSlug ) ) {
+				return false;
+			}
```

Comment:
This selector has the potential to return `undefined` while loading: https://github.com/google/site-kit-wp/blob/eff6e9ad895e350814334ba1732d7af8417f9ea3/assets/js/googlesitekit/modules/datastore/modules.js#L1043-L1046

We should protect against that by checking specifically for `undefined` here and returning `undefined` if it's loading.

Reply:
Good catch. Updated. Thanks 👍